deepchecks/nlp/utils/text_properties.py

```diff
@@ -0,0 +1,162 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the text properties for the NLP module."""
+import string
+from typing import Dict, List, Sequence
+
+import numpy as np
+
+__all__ = ['calculate_default_properties']
+
+
+def text_length(raw_text: Sequence[str]) -> List[int]:
+    """Return list of integers of text lengths."""
+    return [len(text) for text in raw_text]
+
+
+def word_length(raw_text: Sequence[str]) -> List[int]:  # Not yet used as returns list per sample and not number
+    """Return list of integers of word lengths."""
+    return [len(word) for text in raw_text for word in text.split()]
+
+
+def average_word_length(raw_text: Sequence[str]) -> List[float]:
+    """Return list of floats of average word length."""
+    return [np.mean([len(word) for word in text.split()]) for text in raw_text]
+
+
+def percentage_special_characters(raw_text: Sequence[str]) -> List[float]:
+    """Return list of floats of percentage of special characters."""
+    return [len([c for c in text if c in string.punctuation]) / len(text) for text in raw_text]
+
+
+def get_language_detection() -> callable:
+    """Return a function that returns the language of a text.
+
+    As identifying the language of a text requires the langdetect package, which is not necessary for the rest of the
+    NLP module, we import it only when needed.
+    Therefore, the language property is not given directly, but rather through this function, so that the user will only
+    get an ImportError if they try to use the language property.
+    """
+    try:
+        import langdetect  # pylint: disable=import-outside-toplevel
+        from langdetect import DetectorFactory  # pylint: disable=import-outside-toplevel
+        DetectorFactory.seed = 42
+    except ImportError as e:
+        raise ImportError(
```

Comment:
maybe we want these as warnings? so you'll get the properties you can get, and get warnings about ones that you need to do stuff so you can get them.

Reply:
Hm. But then you'll need to rerun it all... Weird as a warning in my opinion. You can just ignore these if you want...

Reply:
How do you ignore an error?